File: Programmers/Java/N개의 최소공배수.java
class Solution {
    
    static private int gcd(int a, int b) {
       return b == 0 ? a : gcd(b,a%b);
    }
    
    static private int lcm(int a, int b) {
        return a/gcd(a,b)*b;
    }
    
    public int solution(int[] arr) {
        int ans = 1;
        for(int num : arr)
            ans = lcm(ans,num);
        return ans;
    }
}

File: Programmers/Python/N개의 최소공배수.py
def gcd(n,m):
    return n if m == 0 else gcd(m,n%m)

def lcm(n,m):
    return n//gcd(n,m)*m

def solution(arr):
    ans = 1
    for num in arr:
        ans = lcm(num,ans)
    return ans
